src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -60,14 +71,31 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 			target = localize('executionPlanPropertiesEdgeOperationName', "Edge");
 		}
 
-		const titleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
-		this._bottomOperationNameContainer.innerText = titleText;
-		this._bottomOperationNameContainer.title = titleText;
+		this._rightTitleText = localize('executionPlanComparisonPropertiesRightOperation', "Right operation: {0}", target);
+		this._bottomTitleText = localize('executionPlanComparisonPropertiesBottomOperation', "Bottom operation: {0}", target);
+		this._bottomOperationNameContainer.innerText = this._bottomTitleText;
+		this._bottomOperationNameContainer.title = this._bottomTitleText;
 		this.addDataToTable();
 	}
 
+	public updatePropertyContainerTitles(orientation: 'horizontal' | 'vertical'): void {
+		if (orientation === 'horizontal') {
+			this._topOperationNameContainer.innerText = this._topTitleText;
+			this._topOperationNameContainer.title = this._topTitleText;
+			this._bottomOperationNameContainer.innerText = this._bottomTitleText;
+			this._bottomOperationNameContainer.title = this._bottomTitleText;
+		}
+		else {
+			this._topOperationNameContainer.innerText = this._leftTitleText;
+			this._topOperationNameContainer.title = this._leftTitleText;
+			this._bottomOperationNameContainer.innerText = this._rightTitleText;
+			this._bottomOperationNameContainer.title = this._rightTitleText;
+		}
 
-	public addDataToTable() {
+		this.addDataToTable(orientation);
+	}
+
+	public addDataToTable(orientation: 'vertical' | 'horizontal' = 'vertical') {
```

Comment:
Why is the orientation tied to adding data to the table? It seems like that should be a completely separate operation generally.

Reply:
That method was named terribly and it's used to refresh table data, which is why this same method is called when we sort the properties table alphabetically. I renamed the method to make it less confusing.